app/services/bulk_republisher.rb

```diff
@@ -41,11 +43,31 @@ def republish_all_documents_with_publicly_visible_editions_with_html_attachments
     republish_by_document_ids(document_ids)
   end
 
+  def republish_all_by_type(type)
+    begin
+      content_type = type.constantize
+    rescue NameError
+      raise "Unknown content type #{content_type}\nCheck the GOV.UK developer documentation for a list of acceptable document types: https://docs.publishing.service.gov.uk/manual/republishing-content.html#whitehall"
+    end
+
+    if non_editionable_content_types.include?(type)
+      republish_non_editionable_content_types(content_type.all)
```

Comment:
I think the benefits of using `find_each` in `republish_non_editionable_content_types` are lost by calling `all` here. Could you just pass in the model instead? I think the other method will continue to work the same

Reply:
Ah good point! I'd used the original implementation from the Rake task but didn't realise we could get rid of the `all` here.